### PR TITLE
CLOUDP-257200: Add support to generate an yaml FOAS

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ CLOUDP-257200

- add the option to generate FOAS in YAML 

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: tools/cli/go.mod [hunk 1]
```diff
@@ -9,6 +9,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	github.com/tufin/oasdiff v1.10.17
 	go.uber.org/mock v0.4.0
+	gopkg.in/yaml.v3 v3.0.1
```
Comment (andreaangiolillo) [inline tools/cli/go.mod:12]: is there a reason why we want to add support for yaml? The reason why we decided to support only JSON was for some limitations with the tools that we use. I don't remember now in detail, but I think that the external references may not work very for YAML. I need to look at the investigation tickets but so far I've found only this [note](https://docs.google.com/document/d/1gQbM_Wv_hMBospC__iR69sG-c2zPsL5YbCvI9esbyss/edit#bookmark=id.wxhrt6xcvr7e)

Comment (blva) [reply tools/cli/go.mod:12]: we default to json indeed, but some tools may need yaml (SDK gen tools use yaml), so we can just ensure to always generate both but defualt to json

Comment (blva) [reply tools/cli/go.mod:12]: we preferred json because of the versioning annotations and date handling AFAIK

**Review by andreaangiolillo — APPROVED** [review]
LGTM

**Pull request merged**